TestApp/Navigation/Sources/Navigation/Modal/Dismissal/ExternallyDismissableModal.swift

```diff
@@ -0,0 +1,45 @@
+import SwiftUI
+
+// A wrapper that makes it possible to dismiss a modally presented view using `ModalDismissalHandlers`
+struct ExternallyDismissableModal<Content: View>: View {
+
+    @EnvironmentObject var dismissalHandlers: ModalDismissalHandlers
+    @StateObject var handler: ModalDismissalHandler
+    let isPresented: Binding<Bool>
+    let content: (Binding<Bool>) -> Content
+
+    init(
+        _ kind: ModalDismissalHandler.Kind,
+        isPresented: Binding<Bool>,
+        content: @escaping (Binding<Bool>) -> Content
+    ) {
+        self._handler = StateObject(wrappedValue: ModalDismissalHandler(kind: kind))
+        self.isPresented = isPresented
+        self.content = content
+    }
+
+    var body: some View {
+        content(
+            Binding(
+                get: {
+                    isPresented.wrappedValue && handler.presentationOverride
+                },
+                set: { isPresented, transaction in
+                    if isPresented == false {
+                        self.isPresented.transaction(transaction).wrappedValue = false
+                        handler.presentationOverride = true
+                    }
+                }
+            )
+        )
+        .onChange(of: isPresented.wrappedValue) { isPresented in
+            // The `isPresented` binding's setter above is not triggered during presentation, only dismissal.
```

Comment:
What are the main reasons why we need the wrapper? Was the native `onDismiss` handling missing something we need? Or is it to avoid repeating `onDismiss` for all specified modifiers on a single screen?

Reply:
This is all for deeplinks. We need to be able to trigger the dismissal from some other location than the actual view presenting this. Also cascading dismissal.